src/coreclr/jit/compiler.h

```diff
@@ -9031,6 +9111,11 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 
         if (type == UnrollKind::Memmove)
         {
+#if defined(FEATURE_SIMD) && defined(TARGET_XARCH)
+            // Enable AVX512 support for Memmove:
+            maxRegSize = compOpportunisticallyDependsOn(InstructionSet_AVX512F) ? ZMM_REGSIZE_BYTES : maxRegSize;
+#endif
+
```

Comment:
You're adding this here because you're only implementing Memmove and not Memset?

Why not remove this, then above, change:

```
            // TODO-XARCH-AVX512: Consider enabling this for AVX512 where it's beneficial
            maxRegSize = min(maxRegSize, YMM_REGSIZE_BYTES);
```
to:
```
    if (type != UnrollKind::Memmove)
    {
            // TODO-XARCH-AVX512: Consider enabling this for AVX512 where it's beneficial
            maxRegSize = min(maxRegSize, YMM_REGSIZE_BYTES);
    }
```

Reply:
Good idea, addressed